crates/ruff_python_codegen/src/stylist.rs

```diff
@@ -102,11 +102,11 @@ pub enum Quote {
     Double,
 }
 
-impl From<ruff_python_parser::QuoteStyle> for Quote {
-    fn from(value: ruff_python_parser::QuoteStyle) -> Self {
+impl From<ruff_python_ast::str::QuoteStyle> for Quote {
+    fn from(value: ruff_python_ast::str::QuoteStyle) -> Self {
         match value {
-            ruff_python_parser::QuoteStyle::Double => Self::Double,
-            ruff_python_parser::QuoteStyle::Single => Self::Single,
+            ruff_python_ast::str::QuoteStyle::Double => Self::Double,
+            ruff_python_ast::str::QuoteStyle::Single => Self::Single,
         }
     }
 }
```

Comment:
We could do, but it would require making the `ruff_python_literal` crate have a dependency on the `ruff_python_ast` crate (or vice versa), due to this:

https://github.com/astral-sh/ruff/blob/965adbed4b56a76d6cce02534e47c4d7a2a9cd32/crates/ruff_python_codegen/src/stylist.rs#L123-L130

If we used `ruff_python_ast::str::QuoteStyle` there instead of `ruff_python_codegen::stylist::Quote`, we wouldn't be able to have the implementation of that trait in the `ruff_python_codegen` crate, since the `QuoteStyle` enum comes from the `ruff_python_literal` crate. `ruff_python_literal` does not currently depend on `ruff_python_ast` (nor vice versa).